src/validation.cpp

```diff
@@ -2245,6 +2248,173 @@ bool BitcoindRPCCheck(const bool init)
     return true;
 }
 
+bool IsValidPeginWitness(const CScriptWitness& pegin_witness) {
+
+    // Format on stack is as follows:
+    // 1) txid - txid of parent chain transaction - TODO-PEGIN unneeded?
```

Comment:
I think we should drop the txid, it's cheap to compute and it's an extra thing to validate. Especially now that the transaction is serialized entirely without that 520-byte splitting frustration.

Reply:
Note that computing the txid does not require understanding Bitcoin transactions. You just have to double-hash the serialized form.

Reply:
The txid is also already include in the prevout. We should drop it.

Reply:
as well as the nOut*

Reply:
Agreed.